addons/sale_coupon/models/sale_order.py

```diff
@@ -141,20 +141,81 @@ def _get_reward_values_discount_percentage_per_line(self, program, line):
         discount_amount = line.product_uom_qty * line.price_reduce * (program.discount_percentage / 100)
         return discount_amount
 
-    def _get_reward_values_discount(self, program):
-        if program.discount_type == 'fixed_amount':
-            taxes = program.discount_line_product_id.taxes_id
-            if self.fiscal_position_id:
-                taxes = self.fiscal_position_id.map_tax(taxes)
+    def _get_max_reward_values_per_tax(self, program, taxes):
```

Comment:
A bit misleading, I would expect to receive the reward values (as per the method name), but this is just about returning the total amount of all the lines (discount lines included) for the given taxes, except the lines of the current program.